### ISSUE TITLE
no member named 'check' in namespace 'eosio'

### ISSUE THREAD

**Issue by ekkis** [EOSIO/eosio.cdt#532]
I'm finding it practically impossible to get any work done with v1.6.1 and I would dearly love to downgrade but the 1.5.0 version I was previously using is no longer available from brew... so I just keep plodding on...

```bash
$ cat > t.cpp <<EOF
#include <eosio/system.hpp>
EOF
$ eosio-cpp t.cpp
```
begets:

> In file included from /Users/ekkis/dev/eos/t.cpp:1:
> In file included from /usr/local/Cellar/eosio.cdt/1.6.1/opt/eosio.cdt/bin/../include/eosiolib/contracts/eosio/system.hpp:6:
> /usr/local/Cellar/eosio.cdt/1.6.1/opt/eosio.cdt/bin/../include/eosiolib/contracts/eosio/../../core/eosio/time.hpp:157:20: error: 
>       no member named 'check' in namespace 'eosio'
>             eosio::check( std::numeric_limits<uint32_t>::max() - slot >= 1, "block timestamp o...
>             ~~~~~~~^
> Warning, empty ricardian clause file
> Warning, empty ricardian clause file
> 1 error generated.
> Error while processing /Users/ekkis/dev/eos/t.cpp.
> abigen error

god almighty.  I'm including system.hpp because they bloody removed `now()` (which I used to write timestamps) so now I have to call `current_time_point()`, for which I need to include system.hpp

so this is super simple but it won't compile.  what do I need?

**Comment by ekkis** [comment]
ok, so the issue is that "check" is defined in `eosio.hpp` so that must be included and included first.  really, `system.hpp` should include `eosio.hpp` since it depends on it, so I'll leave this open as a reminder to someone that `system.hpp` needs to be fixed

for those suffering from the lack of `now()`, here's how I've redefined it:
```c++
#include <eosio/eosio.hpp>
#include <eosio/system.hpp>
uint32_t now() {
    return (uint32_t)(eosio::current_time_point().sec_since_epoch());
}
```

**Comment by jeffreyssmith2nd** [comment]
Fixed in #559

**Comment by ekkis** [comment]
awesome!
